### PR TITLE
[ENH] Address deprecated references in datatypes module

### LINKED ISSUE #2049: [ENH] Address deprecated references in datatypes module
I get this future warnings, mildly annoying and probably easy to fix? @fkiraly 

image

### PR DESCRIPTION
**Pull request by gchhablani**
#### Reference Issues/PRs
Fixes #2049.

#### What does this implement/fix? Explain your changes.

In this fix, I:
- Removed all references of `pd.Int64Index` across all files.
- Updated the tests so verify against `pd.Index` and `dtype=='int64'`.
- Fixed all instantiations from `pd.Int64Index(...)` to `pd.Index(..., dtype='int64')`.
 
#### Does your contribution introduce a new dependency? If yes, which one?
NA

#### What should a reviewer concentrate their feedback on?

- Whether the fix looks okay.
- If there are any missing entries that should be changed.
- If there are other places where the change affects the code-base.

#### Any other comments?

- There were two places where docstrings were missing because of which `pre-commit` was failing. I have added the docstrings as well. If it needs to be added in a separate PR, please let me know.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- ~I've added unit tests and made sure they pass locally~

### CODE REVIEW

**Comment by gchhablani** [comment]
@fkiraly Thank a lot! I missed that.

**Comment by khrapovs** [comment]
This PR looks quite promising actually. It covers much more that what I did here: https://github.com/alan-turing-institute/sktime/pull/2339. It should definitely be revitalized.

**Comment by gchhablani** [comment]
@khrapovs @fkiraly I can get back onto it if that is still a possibility?
Wdyt?

**Comment by khrapovs** [comment]
@gchhablani What do you mean "a possibility"? The issue is still open. Warnings are all over the place. Your PR seem to be resolving a lot of them. So why did you stop?

**Comment by Ennosigaeon** [comment]
I would also appreciate getting this PR merged. The warnings are quite "verbose"...

**Comment by khrapovs** [comment]
I have partially addressed those warnings in https://github.com/alan-turing-institute/sktime/pull/2339, but this PR seem to be more comprehensive.

File: sktime/forecasting/base/tests/test_fh.py [hunk 1]
```diff
@@ -138,8 +144,8 @@ def test_check_fh_values_duplicate_input_values(arg):
         ForecastingHorizon(arg)
 
 
-GOOD_ABSOLUTE_INPUT_ARGS = (
-    pd.Int64Index([1, 2, 3]),
+GOOD_INPUT_ARGS = (
```
Comment (Ennosigaeon) [inline sktime/forecasting/base/tests/test_fh.py:147]: You also have to rename the variable in line 151

**Comment by khrapovs** [comment]
This PR can be closed. The issue https://github.com/alan-turing-institute/sktime/issues/2332 is resolved. Sorry, @gchhablani, these warnings were too annoying.

**Comment by fkiraly** [comment]
yes, I think they are, and this is now heavily conflicting.

Thanks a lot for your work, @gchhablani.
Feel free to add yourself as a `code` contributor since @khrapovs' work was partially based on yours.

**Pull request closed without merging**